scripts/index_aster_lpdaac.py

```diff
@@ -0,0 +1,640 @@
+
+    # Find a file which has the specified bands of this product
+    file_path = None
+    for file_path_ in file_paths:
+        try:
+            _ = selected_bands(file_path_, product)
+        except AssertionError:
+            pass
+        else:
+            file_path = file_path_
+            break
+
+    if file_path:
+        measurements = raster_to_measurements(file_path, product)
+        for measure in measurements:
+            measure.pop('path')  # This is not needed here
+        print_dict(measurements)
+        product_def = generate_lpdaac_defn(measurements, product)
+        print_dict(product_def)
+
+        print(index)
+        product = index.products.from_doc(product_def)
+        print(product)
+        indexed_product = index.products.add(product)
+        print(indexed_product)
+    else:
+        logging.error("No file found having the specified bands of this product: %s", product)
+
+
+@cli.command()
+@click.argument('path')
+@click.option('--product', help='Which ASTER product? vnir, swir, or tir')
+@click.pass_obj
+def index_data(index, path, product):
+    file_paths = find_lpdaac_file_paths(Path(path))
+    print(file_paths)
+
+    resolver = Doc2Dataset(index)
+    for file_path in file_paths:
+
+        bands = selected_bands(file_path, product)
+        if bands:
+            vrt_path = vrt_file_path(file_path, product)
+
+            if vrt_path.exists():
+
+                doc = generate_lpdaac_doc(file_path, product)
+                print_dict(doc)
+                dataset, err = resolver(doc, vrt_path.as_uri())
+
+                print(dataset)
+                if err is not None:
+                    logging.error("%s", err)
+                try:
+                    index.datasets.add(dataset)
+                except Exception as e:
+                    logging.error("Couldn't index %s", file_path)
+                    logging.exception("Exception", e)
+                else:
+                    with open(yaml_file_path(file_path, product), 'w') as yaml_file:
+                        yaml.safe_dump(doc, yaml_file)
+
+            else:
+                logging.error("VRT file not found: %s", vrt_path)
+        else:
+            logging.error("File does not have bands of this product: %s", file_path)
+
+
+def vrt_file_path(file_path, product):
+    return file_path.with_name(f'{file_path.stem}_{product}.vrt')
+
+
+def yaml_file_path(file_path, product):
+    return file_path.with_name(f'{file_path.stem}_{product}.yaml')
+
+
+def print_dict(doc):
+    print(json.dumps(doc, indent=4, sort_keys=True, cls=NumpySafeEncoder))
+
+
+def find_lpdaac_file_paths(path: Path):
+    """
+    Return a list of hdf file path objects.
+
+    :param path:
+    :return: A list of path objects.
+    """
+    file_paths = []
+    for afile in path.iterdir():
+        if afile.suffix == '.hdf' and afile.stem[:7] == 'AST_L1T':
+            file_paths.append(afile)
+    return file_paths
```

Comment:
Should probably be a generator function. If the caller needs a realised `list`, it can do it itself.

Reply:
Yep done